source/includes/write/delete.php

```diff
@@ -0,0 +1,34 @@
+<?php
+require 'vendor/autoload.php';
+
+$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
+$client = new MongoDB\Client($uri);
+
+// start-db-coll
+$collection = $client->sample_restaurants->restaurants;
+// end-db-coll
+
+// Deletes a document that has a "name" value of "Ready Penny Inn"
+// start-delete-one
+$result = $collection->deleteOne(['name' => 'Ready Penny Inn']);
+// end-delete-one
+
+// Deletes documents that have a "borough" value of "Brooklyn"
+// start-delete-many
+$result = $collection->deleteMany(['borough' => 'Brooklyn']);
+// end-delete-many
+
+// Deletes matching documents and attaches a comment to the operation
+// start-delete-options
+$result = $collection->deleteMany(
+    ['name' => new MongoDB\BSON\Regex('Mongo', '')],
```

Comment:
If you're not using an options (i.e. regex flags), you can omit the optional second parameter:

```suggestion
    ['name' => new MongoDB\BSON\Regex('Mongo')],
```

----

Do other language docs use the BSON regex type or the [`$regex`](https://www.mongodb.com/docs/manual/reference/operator/query/regex/) query operator? If we want to ensure consistency across all language examples, the operator might be the way to go.

The server docs [note differences](https://www.mongodb.com/docs/manual/reference/operator/query/regex/#-regex-vs.--pattern--syntax) and point out where you'd need the BSON type; however, they're also written from the perspective of the shell where Javascript's [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) (object or literal) is used and then serialized to a BSON type over the wire. RegExp has some of its own limitations (as do native regex types in other languages). In contrast, PHP has no native regex object and the `MongoDB\BSON\Regex` object is just a container for two strings (pattern and options), which serialize as-is to a BSON regex type (i.e. we can express anything the [server supports](https://bsonspec.org/spec.html)).

Reply:
Good point, it's kind of a mix - the [PyMongo](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/write/delete/#customize-the-delete-operation) and [Go](https://www.mongodb.com/docs/drivers/go/current/fundamentals/crud/read-operations/query-document/#example-4) docs use $regex, and the [Java sync](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/crud/query-document/#evaluation-operators) and [Reactive Streams](https://www.mongodb.com/docs/languages/java/reactive-streams-driver/upcoming/write/write-delete-documents/#example) docs use helper methods, for ex. So it wouldn't really help with consistency to change this example to use the $regex operator.